pkg/controller/job/job_controller.go

```diff
@@ -902,20 +902,30 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 
 	var manageJobErr error
 
+	// This is the starting point for evaluating the end state of the Job.
+	// Note that we need to order evaluations since a Job could satisfy multiple criteria at the same time in some cases:
+	// 1. Evaluate the pre-existing SuccessCriteriaMet and FailureTarget to respect the previous reconcile results.
+	// 2. Evaluate failure scenarios
+	// 3. Evaluate success scenarios
```

Comment:
Yeah, I don't think there is any technical reason for the different places of promoting the conditions. So I think aligning that would be a great follow up cleanup. Since success policy is still beta while failure policy is ga, maybe we could align into that order of things, but I'm ok either way   WDYT?

Reply:
@atiratree Does that mean like the comments below?

```
	// This is the starting point for evaluating the end state of the Job.
	// Note that we need to order evaluations since a Job could satisfy multiple criteria at the same time in some cases:
	// 1. Evaluate the pre-existing SuccessCriteriaMet and FailureTarget to respect the previous reconcile results.
	// 2. Evaluate failure scenarios
	// 3. Evaluate success scenarios
	// 4. Evaluate finishedCondition, then transform FailureTarget to Failed and SuccessCriteriaMet to Complete.
```

Reply:
Ok, we can revisit this with the success policy (or even separately). It would be nice to unify these to decrease the cognitive load. Let's at least document it for now.

Reply:
> FailureTarget (transformed to Failed here and later)
SuccessCriteriaMet (transformed to Complete later)

Ah, I found the differences. FailureTarget and SuccessPolicy are transformed in different place

Reply:
```
[...]
	// 1. Evaluate the pre-existing SuccessCriteriaMet and FailureTarget to respect the previous reconcile results. FailureTarget is transformed to Failed condition.
[...]
```

@atiratree How about this?

Reply:
That makes sense.
It looks like both are valuable. So, let me refine `1` and add `4`.

Reply:
Done. This is the differeneces: https://github.com/kubernetes/kubernetes/compare/72791344060005fc4b99016893c28e44953e0f08..5dda60ee4ef4769dc63dcd312a5cbb0c3023df29

Reply:
@atiratree For refactoring the SuccessPolicy transformation, I would like to work on it after the 1.32 code freeze since I think the status transition refactoring should be done carefully.

Reply:
sure, there is no rush with that